geoips/plugins/modules/readers/abi_netcdf.py

```diff
@@ -509,6 +510,94 @@ def call(fnames, metadata_only=False, chans=None, area_def=None, self_register=F
     """
     Read ABI NetCDF data from a list of filenames.
 
+    Parameters
+    ----------
+    fnames : list
+        * List of strings, full paths to files
+    metadata_only : bool, default=False
+        * Return before actually reading data if True
+    chans : list of str, default=None
+        * List of desired channels (skip unneeded variables as needed).
+        * Include all channels if None.
+    area_def : pyresample.AreaDefinition, default=None
+        * Specify region to read
+        * Read all data if None.
+    self_register : str or bool, default=False
+        * register all data to the specified dataset id (as specified in the
+          return dictionary keys).
+        * Read multiple resolutions of data if False.
+
+    Returns
+    -------
+    dict of xarray.Datasets
+        * dictionary of xarray.Dataset objects with required Variables and
+          Attributes.
+        * Dictionary keys can be any descriptive dataset ids.
+
+    See Also
+    --------
+    :ref:`xarray_standards`
+        Additional information regarding required attributes and variables
+        for GeoIPS-formatted xarray Datasets.
+    """
+    all_metadata = readers.concatenate_metadata(
+        [call_single_time([x], metadata_only=True)["METADATA"] for x in fnames]
+    )
+    if metadata_only:
+        return all_metadata
+    start_times = [x.start_datetime for x in all_metadata]
+    times = list(set(start_times))
+    import collections
+
+    ingested_xarrays = collections.defaultdict(list)
+    for time in times:
+        scan_time_files = [x.start_datetime == time for x in all_metadata]
+        data_dict = call_single_time(
+            np.array(fnames)[scan_time_files],
+            metadata_only=metadata_only,
+            chans=chans,
+            area_def=area_def,
+            self_register=self_register,
+        )
+        for (
+            dname,
+            dset,
+        ) in data_dict.items():
+            ingested_xarrays[dname].append(dset)
+
+    if len(times) == 1:
+        # No need to stack if we are only reading in one scan time
+        # This is likely temporary to maintain backwards compatibility
+        return data_dict
+
+    import xarray
+
+    # merged_dset = xarray.Dataset()
+    # Now that we've ingested all scan times, stack along time dimension
+    dict_xarrays = {}
+    for dname, list_xarrays in ingested_xarrays.items():
+        if dname == "METADATA":
+            continue
+        merged_dset = xarray.concat(list_xarrays, dim="time_dim")
```

Comment:
@jsolbrig @cpcamacho says that he is surprised you are not commenting on "time_dim" because he thought you didn't like it :)